File: java/bitcoin/Number/NumberFactory.java
import java.util.Random;
import java.math.BigInteger;

public abstract class NumberFactory
{
  public abstract Number zero();
  public abstract Number one();
  public abstract Number fromBytes(int signum, byte[] val); // input big-endian
  public abstract Number random(int numBits, Random rnd);   // U(0, 2^nb - 1)
  public abstract Number fromBigInteger(BigInteger n);
}

File: c#/bitcoin/Number/NumberFactory.cs
using System;
using System.Numerics;
using System.Security.Cryptography;
using Random = System.Security.Cryptography.RandomNumberGenerator;

public abstract class NumberFactory {
  public abstract Number Zero();
  public abstract Number One();
  public abstract Number FromBytes(int sign, byte[] val); // input big-endian
  public abstract Number Random(int numBits, Random rnd); // uniform
  public abstract Number FromBigInteger(BigInteger n);      
}
